terraform/cloud-platform-components/resources/fluentd-es/templates/fluentd-es.yaml

```diff
@@ -0,0 +1,118 @@
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: {{ .Release.Name }}
+  namespace: {{ .Values.namespace }}
+  labels:
+    k8s-app: fluentd-es
+    kubernetes.io/cluster-service: "true"
+    addonmanager.kubernetes.io/mode: Reconcile
+---
+kind: ClusterRole
+apiVersion: rbac.authorization.k8s.io/v1
+metadata:
+  name: {{ .Release.Name }}
+  labels:
+    k8s-app: fluentd-es
+    kubernetes.io/cluster-service: "true"
+    addonmanager.kubernetes.io/mode: Reconcile
+rules:
+- apiGroups:
+  - ""
+  resources:
+  - "namespaces"
+  - "pods"
+  verbs:
+  - "get"
+  - "watch"
+  - "list"
+---
+kind: ClusterRoleBinding
+apiVersion: rbac.authorization.k8s.io/v1
+metadata:
+  name: {{ .Release.Name }}
+  labels:
+    k8s-app: fluentd-es
+    kubernetes.io/cluster-service: "true"
+    addonmanager.kubernetes.io/mode: Reconcile
+subjects:
+- kind: ServiceAccount
+  name: {{ .Release.Name }}
+  namespace: {{ .Values.namespace }}
+  apiGroup: ""
+roleRef:
+  kind: ClusterRole
+  name: {{ .Release.Name }}
+  apiGroup: ""
+---
+apiVersion: apps/v1
+kind: DaemonSet
+metadata:
+  name: fluentd-es-{{ .Values.version }}
```

Comment:
When you create a new chart using `helm` (`helm create foo`) you'll see that the resources have a number of labels defined:
```
  name: {{ include "foo.fullname" . }}
  labels:
    app: {{ include "foo.name" . }}
    chart: {{ include "foo.chart" . }}
    release: {{ .Release.Name }}
    heritage: {{ .Release.Service }}
```
I've not found any documentation on how these are used by `helm` but it seems they are used. `nginx-ingress` and `kube-prometheus` define them on all resources. I think we should do the same?

Reply:
Where are you referencing this from?

Reply:
Try `helm create newchart` and look at the templates.
Also [example from `nginx-ingress`](https://github.com/helm/charts/blob/master/stable/nginx-ingress/templates/controller-deployment.yaml#L4-L11)